src/app/backend/deploy.go

```diff
@@ -231,3 +246,65 @@ func getLabelsMap(labels []Label) map[string]string {
 
 	return result
 }
+
+// Deploys an app based on the given yaml or json file.
+func DeployAppFromFile(spec *AppDeploymentFromFileSpec, client *client.Client) error {
+	cmd := create.NewCmdCreate(cmdutil.NewFactory(nil), os.Stdout)
+	return RunCreate(cmdutil.NewFactory(nil), cmd, os.Stdout, spec)
+}
+
+// Validates and runs the given command
+func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, spec *AppDeploymentFromFileSpec) error {
+	schema, err := f.Validator(cmdutil.GetFlagBool(cmd, "validate"), cmdutil.GetFlagString(cmd, "schema-cache-dir"))
+	if err != nil {
+		return err
+	}
+	cmdNamespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	mapper, typer := f.Object()
+	reader := strings.NewReader(spec.Content)
+
+	r := kubectlResource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		Schema(schema).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		Stream(reader, spec.Name).
+		Flatten().
+		Do()
+
+	count := 0
+	err = r.Visit(func(info *kubectlResource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if err := createAndRefresh(info); err != nil {
+			return cmdutil.AddSourceToErr("creating", info.Source, err)
+		}
+
+		count++
+		cmdutil.PrintSuccess(mapper, true, out, info.Mapping.Resource, info.Name, "created")
```

Comment:
Use logging not cmdutil here. Basically, use as little from kubectl as possible :)